RxSwift/Observables/Implementations/Materialize.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  Materialize.swift
+//  Rx
+//
+//  Created by sergdort on 08/03/2017.
+//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
+//
+
+import Foundation
+
+public protocol EventType {
+    associatedtype E
+    func asEvent() -> Event<E>
+}
+
+extension Event : EventType {
+    public typealias E = Element
+    public func asEvent() -> Event<E> { return self }
+}
+
+class Obs<E>: ObserverType {
+    func on(_ event: Event<E>) {
+        
+    }
+}
+
+fileprivate final class MaterializeSink<Element>: Disposable, ObserverType {
+    private let _observer: AnyObserver<Event<Element>>
+    private let _cancel: Cancelable
+    private var _disposed: Bool
+    
+    init(observer: AnyObserver<Event<Element>>, cancel: Cancelable) {
```

Comment:
Not sure if it's possible to use  `ObserverType` with generic constraint instead of `AnyObserver`